bpf/kmesh/probes/metrics.h

```diff
@@ -45,21 +44,38 @@ static inline void construct_metric_key(struct bpf_sock *sk, __u8 direction, str
         if (sk->family == AF_INET) {
             key->src_ip.ip4 = sk->src_ip4;
             key->dst_ip.ip4 = sk->dst_ip4;
-        } else {
+        }
+        if (sk->family == AF_INET6) {
             bpf_memcpy(key->src_ip.ip6, sk->src_ip6, IPV6_ADDR_LEN);
             bpf_memcpy(key->dst_ip.ip6, sk->dst_ip6, IPV6_ADDR_LEN);
         }
-        key->dst_port = bpf_ntohl(sk->dst_port);
-    } else {
+        key->dst_port = bpf_ntohs(sk->dst_port);
+    }
+    if (direction == INBOUND) {
         if (sk->family == AF_INET) {
             key->src_ip.ip4 = sk->dst_ip4;
             key->dst_ip.ip4 = sk->src_ip4;
-        } else {
+        }
+        if (sk->family == AF_INET6) {
```

Comment:
Why change from `else` to this? Are there other situations? If so, do we need to handle the exceptions?